src/main/io/osd.c

```diff
@@ -395,7 +395,10 @@ static void osdDrawSingleElement(uint8_t item)
     case OSD_HOME_DIR:
         if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
             if (GPS_distanceToHome > 0) {
-                const int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+				int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+				// If MAG_Heading is not available, yaw starts with zero in a random direction. With GPS course over ground this can be roughly corrected
+				if (sensors(SENSOR_GPS)) h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(gpsSol.groundCourse);
```

Comment:
Betaflight coding standards are to always use a block with branching  operators (except for switch).
The reason for this is maintainability. Without a block it is very easy for somebody to add an extra statement that is meant to be inside the branch, but isn't because there is no block.

Also, I think instead of ignoring the actual heading just for this single case if GPS is ignored, which is confusing since all other places will still be reporting the true yaw instead of a GPS based assumption, the GPS heading should be taken into account in `flight/imu.c`. It actually already is, in the case of fixed wing craft, so instead of this change you should probably just change your model to a fixed wing craft to get the desired behaviour: https://github.com/betaflight/betaflight/blob/a618de9e8e59611ae0a484ea244ec6630d7411f7/src/main/flight/imu.c#L431

Reply:
"Branching operators" - checked.
A suggestion misled me. Now the override of the direction should only happen if GPS is up and running and no MAG is detected.
Fixed wing - thanks for that idea. But isn't the mixer changed by this model type and does not fit to a quad?